src/proto/FunctionRpc.proto

```diff
@@ -117,13 +117,35 @@ message WorkerInitRequest {
 
 // Worker responds with the result of initializing itself
 message WorkerInitResponse {
-  // Version of worker
+  // NOT USED
+  // TODO: Remove from protobuf during next breaking change release
   string worker_version = 1;
+
   // A map of worker supported features/capabilities
   map<string, string> capabilities = 2;
 
   // Status of the response
   StatusResult result = 3;
+
+  // Worker metadata captured for telemetry purposes
+  WorkerMetadata worker_metadata = 4;
+}
+
+message WorkerMetadata {
+  // The runtime/stack name
+  string runtime_name = 1;
+
+  // The version of the runtime/stack
+  string runtime_version = 2;
+
+  // The version of the worker or worker SDK
```

Comment:
Do we nede the "Or worker SDK" here? If we are going to store the primary worker package version and other packages like SDK, Shouldn't we define some format for all those (ex: a pipe(|`) or comma(,) separated string.

If we do not have this format, we should store only the primary worker package version here. Any other additional info (like the Worker SDK package which is used for build purpose) can go in the custom_properties.

Reply:
For dotnet isolated, worker is the https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker/ package which is needed for running the isolated function app. https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker.Sdk/ is used for build time stuff (generating function meta data, source gen, analyzer etc..)

Reply:
This is because worker_version might mean something different for each worker, but I can remove "or sdk" from the comment if it is confusing. Any additional info should go into `custom_properties`